lib/ruby_lsp/listeners/discover_tests.rb

```diff
@@ -0,0 +1,83 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Listeners
+    class DiscoverTests
+      extend T::Helpers
+      abstract!
+
+      include Requests::Support::Common
+
+      DYNAMIC_REFERENCE_MARKER = "<dynamic_reference>"
```

Comment:
If https://github.com/Shopify/ruby-lsp/pull/3298 is merged then we can remove this.